src/Symfony/Component/Form/Extension/Core/DataTransformer/BooleanToStringTransformer.php

```diff
@@ -80,6 +94,19 @@ public function reverseTransform($value)
             throw new TransformationFailedException('Expected a string.');
         }
 
-        return true;
+        if (in_array($value, array_merge($this->defaultTrueValues, array($this->trueValue)), true)) {
+            return true;
+        }
+
+        if (in_array($value, $this->defaultFalseValues, true)) {
+            return false;
+        }
+
+        throw new TransformationFailedException(
+            sprintf(
+                'Unexpected value "%s"!',
```

Comment:
Can you please add the expected values here as well? Right now, the exception message is rather frustrating for a developer hitting this error.